### PR TITLE
Refactor `CudaTeamMember` implementation to avoid `#ifdef __CUDA_ARCH__`

### PR DESCRIPTION
**Pull request by dalg24**
Format is pretty bad but I don't know how to do better.

### CODE REVIEW

File: core/src/Cuda/Kokkos_Cuda_Team.hpp [hunk 1]
```diff
@@ -130,28 +130,20 @@ class CudaTeamMember {
   KOKKOS_INLINE_FUNCTION int league_rank() const { return m_league_rank; }
   KOKKOS_INLINE_FUNCTION int league_size() const { return m_league_size; }
   KOKKOS_INLINE_FUNCTION int team_rank() const {
-#ifdef __CUDA_ARCH__
-    return threadIdx.y;
-#else
-    return 0;
-#endif
+    KOKKOS_IF_ON_DEVICE((return threadIdx.y;))
+    KOKKOS_IF_ON_DEVICE((return 0;))
```
Comment (ldh4) [inline core/src/Cuda/Kokkos_Cuda_Team.hpp:134]: Maybe I'm misunderstanding something here, but if the previous behavior was to `return threadIdx.y` on device and `return 0` on host, shouldn't this be converted to
```c++
    KOKKOS_IF_ON_DEVICE((return threadIdx.y;))
    KOKKOS_IF_ON_HOST((return 0;))
```

File: core/src/Cuda/Kokkos_Cuda_Team.hpp [hunk 2]
```diff
@@ -130,28 +130,20 @@ class CudaTeamMember {
...
   }
 
   KOKKOS_INLINE_FUNCTION int team_size() const {
-#ifdef __CUDA_ARCH__
-    return blockDim.y;
-#else
-    return 1;
-#endif
+    KOKKOS_IF_ON_DEVICE((return blockDim.y;))
+    KOKKOS_IF_ON_DEVICE((return 1;))
```
Comment (ldh4) [inline core/src/Cuda/Kokkos_Cuda_Team.hpp:139]: Same for this one?

**Review by crtrott — CHANGES_REQUESTED** [review]

File: core/src/Cuda/Kokkos_Cuda_Team.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_Team.hpp]: Fixed

**Review by ldh4 — APPROVED** [review]

**Comment by dalg24** [comment]
Retest this please

**Review by masterleinad — APPROVED** [review]
Looks good to me.

**Review by crtrott — APPROVED** [review]

**Pull request merged**